esmvaltool/cmorizers/data/formatters/datasets/esacci_ozone.py

```diff
@@ -0,0 +1,166 @@
+"""ESMValTool CMORizer for ESACCI-OZONE data.
+
+Tier
+    Tier 2: other freely-available dataset.
+
+Source
+
+Last access
+    20250107
+
+Download and processing instructions
+
+    GTO-ECV total column (variable toz)
+    Select the following from the CDS:
+    https://cds.climate.copernicus.eu/datasets/satellite-ozone-v1?tab=download
+
+    Processing Level = "Level 3"
+    Variable = "Atm. mole content of ozone"
+    Vertical aggregation = "Total column"
+    Sensor = "MERGED-UV"
+    Year = select all (1995-2023)
+    Month = select all (1-12)
+    Version = "v2000"
+
+    SAGE-CCI-OMPS (variable o3)
+    Select the following options from the same link:
+    https://cds.climate.copernicus.eu/datasets/satellite-ozone-v1?tab=download
+
+    Processing Level = "Level 3"
+    Variable = "Mole concentration of ozone in air"
+    Vertical aggregation = " Vertical profiles from limb sensors"
+    Sensor = " CMZM (Monthly zonal mean merged concentration product from limb
+            sensors ACE, GOMOS, MIPAS, OMPS, OSIRIS, SAGE-2 and SCIAMACHY)"
+    Year = select all (1984-2022)
+    Month = select all (1-12)
+    Version = "v0008"
+
+    Put all files under a single directory (no subdirectories with years).
+    in ${RAWOBS}/Tier2/ESACCI-OZONE
+
+"""
+import logging
+from datetime import datetime
+import iris
+import iris.util
+from cf_units import Unit
+from pathlib import Path
+from esmvalcore.preprocessor import (
+    concatenate
+)
+from esmvalcore.cmor._fixes.native_datasets import NativeDatasetFix
+
+from ...utilities import (
+    fix_var_metadata,
+    fix_coords,
+    save_variable,
+    set_global_atts
+)
+
+logger = logging.getLogger(__name__)
+
+
+def _convert_units(cubes, short_name, var):
+    """Perform variable-specific calculations."""
+    cube = cubes.extract_cube(var['raw'])
+    if short_name == 'o3':  # Ozone mole fraction
+        r = 8.31446261815324  # Ideal gas constant (J mol-1 K-1)
+        t_cube = cubes.extract_cube('air_temperature')
+        p_cube = cubes.extract_cube('air_pressure')
+
+        air_mol_concentration = p_cube / (r * t_cube)  # mol m-3
+        mixing_ratio = cube / air_mol_concentration
+        cube = mixing_ratio / (1.0 + mixing_ratio)
+        cube.units = 'mol mol-1'
+
+    elif short_name == 'toz':  # Total ozone column (DU or m)
+        cube = cube * 2241.399
+        cube.units = 'DU'
+        cube = cube * 2.24115e-5
+        cube.units = 'm'
```

Comment:
I think the unit conversion is not correct. The total ozone column should be converted from mol m-2 to m (CMOR unit for toz):

```suggestion
    elif short_name == 'toz':  # Total ozone column (m)
        # Convert from mol m-2 to m
        # -------------------------
        # 1e-5 m (gas @ T = 273 K and p = 101325 Pa) ~ 2.69e20 molecules m-2
        # (see https://en.wikipedia.org/wiki/Dobson_unit)
        # 1 m (O3 @ standard conditions) = 1e5 * 2.69e20 molecules m-2
        #                              = 2.69e25 moelcules m-2 / N_A mol m-2
        #                              = 2.69e25 molecules m-2 /
        #                                    (6.022e23 molecules mol-1) mol m-2
        #                              = 44.67 mol m-2
        cube = cube / 44.67
        cube.units = 'm'
```